src/functions-and-arrays.js

```diff
@@ -1,18 +1,112 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(number1 , number2) {
+  return Math.max(number1,number2)
+}
+
+
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+function findLongestWord(words) {
+  let word = '';
+  if (!words.length){
+    return null;
+  } else for (let i = 0; i < words.length; i++) {
+    if (word.length < words[i].length) {
+      word = words[i];
+    }
+  }
+  return word;
+};
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+function sumNumbers (arr){
+  let sum = 0;
+  for (let i = 0 ; i < arr.length ; i++){
+    sum += arr[i]
+  }
+  return sum
+};
+
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+  function sum (mixedArr){
+    let sum = 0;
+    //console.log("array length",mixedArr.length)
+    for (let i = 0 ; i < mixedArr.length ; i++){
+      
+      if (typeof mixedArr[i] === "number"){
+        console.log(sum)
+      sum += mixedArr[i]
+      //console.log("number",sum)
+      } else if ((typeof mixedArr[i] === "boolean") && (mixedArr[i] = true)) {
+        sum += 1;
+       // console.log("boolean",sum)
+      }  else if (typeof mixedArr[i] === "string") {
+      sum += mixedArr[i].length;
+     // console.log("string",sum)
+       }
+    }
+     return sum
+  }
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+function averageNumbers (arr){
+  let sum = 0;
+  if (!arr.length){
+    return null;
+  }
+  for (let i = 0 ; i < arr.length ; i++){
+    sum += arr[i]
+  }
+  return sum / arr.length
```

Comment:
You could also have used your sumNumbers function `return sumNumbers(arr)/arr.length`